web-wallet/src/lib/stores/walletStore.js

```diff
@@ -55,17 +48,35 @@ const getCurrentAddress = () => get(walletStore).currentAddress;
 /** @type {(action: (...args: any[]) => Promise<any>) => Promise<void>} */
 const syncedAction = (action) => sync().then(action).finally(sync);
 
+async function updateAfterSync() {
+  const store = get(walletStore);
+
+  // @ts-expect-error
+  const balance = await walletInstance.getBalance(store.currentAddress);
+
+  set({
+    ...store,
+    balance,
+    syncStatus: { isInProgress: false, current: 0, last: 0, error: null },
+  });
+}
+
 const abortSync = () => {
   syncPromise && syncController?.abort();
   syncPromise = null;
 };
 
-/** @type {() => Promise<void>} */
+/** @type {WalletStoreServices["clearLocalData"]} */
 const clearLocalData = async () => walletInstance?.reset();
 
-/** @type {(wallet: Wallet) => Promise<void>} */
-const clearLocalDataAndInit = (wallet) =>
-  wallet.reset().then(() => init(wallet));
+/** @type {WalletStoreServices["clearLocalDataAndInit"]} */
+const clearLocalDataAndInit = async (wallet, syncFromBlock) => {
+  await wallet.reset();
```

Comment:
Why this change?
Isn't better to just return the "reset & init promise" as before?

Reply:
Will revert the change. I guess I have been experimenting with the function.